api/src/main/java/org/apache/iceberg/ChangelogScanTask.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+/**
+ * A changelog scan task.
+ */
+public interface ChangelogScanTask extends FileScanTask {
+  /**
+   * Returns the operation type (i.e. insert/delete).
+   */
+  ChangelogOperation operation();
+
+  /**
+   * Returns the relative commit order in which the changes must be applied.
+   */
+  int commitOrder();
```

Comment:
what is the scope of the order? is it  the snapshot sequence number? or is it just a relative order for all tasks from one plan result?

Reply:
Snapshot sequence numbers are present only in v2 tables. Also, some records may have different commit orders within one snapshot. Consider a Flink upsert that adds a data file and a position delete file against it. Added records should have a smaller commit order.

Reply:
The idea is that the commit order will start from 0 every time we query for a set of changes.

Reply:
commit implies a single snapshot/transaction to me. what about `changeOrder`?
```
Returns the relative order in which the changes must be applied.
```

Reply:
Change order sounds good to me.

Reply:
Updated.